GitUI/ScriptsEngine/ScriptsManager.ScriptRunner.cs

```diff
@@ -55,98 +55,106 @@ private static bool RunScriptInternal(ScriptInfo script, IWin32Window owner, IGi
-                    bool success = FormProcess.ShowDialog(owner, uiCommands as GitUICommands, argument, uiCommands.GitModule.WorkingDir, null, true, process: command);
-                    if (!success)
-                    {
-                        return false;
+                            if (revisionRef is not null)
+                            {
+                                uiCommands.BrowseRepo.GoToRef(revisionRef, true);
+                            }
+                        }
+
+                        return true;
                     }
 
-                    uiCommands.RepoChangedNotifier.Notify();
-                }
-                else
-                {
-                    if (originalCommand.Equals("{openurl}", StringComparison.CurrentCultureIgnoreCase))
+                    if (!script.RunInBackground)
                     {
-                        OsShellUtil.OpenUrlInDefaultBrowser(argument);
+                        // TODO: Remove downcast from IGitUICommands to GitUICommands after https://github.com/gitextensions/gitextensions/pull/11269
+                        bool success = FormProcess.ShowDialog(owner, uiCommands as GitUICommands, argument, uiCommands.GitModule.WorkingDir, null, true, process: command);
+                        if (!success)
+                        {
+                            return false;
+                        }
+
+                        uiCommands.RepoChangedNotifier.Notify();
                     }
                     else
                     {
-                        // It is totally valid to have a command without an argument, e.g.:
-                        //    Command  : myscript.cmd
-                        //    Arguments: <blank>
-                        new Executable(command, uiCommands.GitModule.WorkingDir).Start(argument ?? string.Empty);
+                        if (originalCommand.Equals("{openurl}", StringComparison.CurrentCultureIgnoreCase))
+                        {
+                            OsShellUtil.OpenUrlInDefaultBrowser(argument);
+                        }
+                        else
+                        {
+                            // It is totally valid to have a command without an argument, e.g.:
+                            //    Command  : myscript.cmd
+                            //    Arguments: <blank>
+                            new Executable(command, uiCommands.GitModule.WorkingDir).Start(argument ?? string.Empty);
+                        }
                     }
-                }
 
-                return true;
+                    return true;
+                }
+                catch (OperationCanceledException)
+                {
+                    MessageBox.Show(owner, TranslatedStrings.ScriptUserCanceledRun, TranslatedStrings.ScriptText, MessageBoxButtons.OK, MessageBoxIcon.Information);
+                    return false;
+                }
```

Comment:
Exceptions are already handled by the single caller `RunScript`, i.e. move it there, please.
```suggestion
                catch (OperationCanceledException)
                {
                    MessageBox.Show(owner, TranslatedStrings.ScriptUserCanceledRun, script.Name, MessageBoxButtons.OK, MessageBoxIcon.None);
                    return false;
                }
```